File: 0153_find_minimum_in_rotated_sorted_array/Solution.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            int mid = (int) (left + (right - left) / 2);

            if (nums[mid] < nums[right]) {
                right = mid;
            } else { // nums[mid] >= nums[right]
                left = mid + 1;
            }
        }

        return nums[left];
    }
}

File: 0153_find_minimum_in_rotated_sorted_array/solution.py
from typing import List


class Solution:
    def findMin(self, nums: List[int]) -> int:
        left: int = 0
        right: int = len(nums) - 1
        while left < right:
            mid: int = left + (right - left) // 2
            if nums[mid] < nums[right]:
                right = mid
            else:  # nums[mid] >= right
                left = mid + 1
        return nums[left]

        